### ISSUE TITLE
Not clear how to add non-generic dependencies to apps

### ISSUE THREAD

**Issue by allgood2386** [hashicorp/otto#228]
So it's not clear to me how to add a dependency such as mysql to a php app. 
In #126 and #133 it seems as if we would edit the vagrant file but this seems like it would be overly broad and unneeded for many php projects. Is there a way at this point to specify an application type (lets say wordpress, laravel, drupal) and have those extra dependencies installed on otto dev? I'm more than happy to build a few of these app configurations I'm just not sure that this is the 'correct' approach for this project and documentation doesn't make it clear at this point.

**Comment by themccallister** [comment]
I was looking into this today as well. 

Does it make sense for the application to be responsible for these dependencies (databases specifically)? With a `simple` infra, this makes sense. Spin up the environment for dev, deploy and build a MySQL server for the node. This approach seems proper and efficient.

However, lets imagine a PHP application that uses the `vpc-public-private` infra. This would create the application with 3 nodes (described here: https://ottoproject.io/docs/foundations/consul.html). This would mean that the database would be installed on each individual node - making the database essentially useless and out of sync.

Basically, there needs to be a way to document/automate this in the Appfile. As well as documentation to configure a resource such as RDS for the application to utilize.

**Comment by allgood2386** [comment]
I hadn't even thought about infra and was only thinking about local work so it's good to think about now. 

Would an app type for sql be appropriate? It could work similarly to the docker app type.

```customization "external-sql" {
    engine = "mysql:5"
    database = "Somename"
    grants = "read"
}```

**Comment by dmfenton** [comment]
I ran into this issue as well. I have a dependency on Postgis which I have running in a container. I needed to install postresql-client-9.4 for a Rails project. Not sure how to make this happen "ottomagically" :stuck_out_tongue:
